okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/RemotingOkHttpCall.java

```diff
@@ -140,6 +149,23 @@ private static ResponseBody buffer(MediaType mediaType, byte[] bodyBytes) {
 
     @Override
     public void enqueue(Callback callback) {
+        ListenableFuture<Listener> listenerFuture = limiter.acquire();
+        Futures.addCallback(listenerFuture, new FutureCallback<Listener>() {
+            @Override
+            public void onSuccess(Listener listener) {
+                enqueue(new RequestCompletion(schedulingExecutor, executionExecutor, callback, listener));
+            }
+
+            @Override
+            public void onFailure(Throwable t) {
+                callback.onFailure(
+                        RemotingOkHttpCall.this,
+                        new IOException(new AssertionError("This should never happen", t)));
```

Comment:
explain a bit more in the message please

Reply:
done